### PR TITLE
feat: can resume and dump from string

### PR DESCRIPTION
**Pull request by yihong0618**
fix: #4 
@matin

### CODE REVIEW

**Comment by matin** [comment]
My main comments:

1. Methods should be called `dumps()` and `loads()` to stay consistent with the same pattern used elsewhere in Python
2. base64 only makes the strings longer. Could you give it a try without that?

**Comment by matin** [comment]
Using base64 is ok. So, to merge:

1. Change method names to `dumps()` and `loads()`
2. Add tests to keep coverage at 100%

**Comment by yihong0618** [comment]
for the reason use base64, for GitHub actions secret or other its easy to copy paste,  especially for some users that do not know much about code.

for 1, its ok for me
for 2, will do it

**Comment by matin** [comment]
I would get rid of `garth.resume_from_string()` and `garth.save_to_string()` and use `garth.client.loads()` and `garth.client.dumps()` instead.

I only added `save()` and `resume()` at the top level to make it easier to read in Jupyter Notebooks, but with your use case, I wouldn't add it.

**Comment by yihong0618** [comment]
OK, will fix my repo and added test tomorrow, thanks again for your great repo, and good night(good day)

**Comment by yihong0618** [comment]
@matin done and left a few review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.9', '3.10', '3.11']
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (yihong0618) [inline .github/workflows/ci.yml:22]: I think we can support 3.8

File: tests/conftest.py [hunk 2]
```diff
@@ -67,9 +67,8 @@ def authed_client(
         client.load(os.environ["GARTH_HOME"])
     except KeyError:
         client.configure(oauth1_token=oauth1_token, oauth2_token=oauth2_token)
-    else:
-        assert client.oauth2_token
-        assert not client.oauth2_token.expired
+    assert client.oauth2_token
+    assert not client.oauth2_token.expired
```
Comment (yihong0618) [inline tests/conftest.py:71]: this should not be in else block to make conv 100%

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,12 +5,11 @@ on:
     branches:
       - main
     tags:
-      - '**'
+      - "**"
   pull_request: {}
 
 env:
   COLUMNS: 150
-  PDM_DEPS: 'urllib3<2,git+https://github.com/adriangb/findpython.git@version-timeout-env'
```
Comment (yihong0618) [inline .github/workflows/ci.yml:13]: pdm 2.9.3  up the min version, so delete this

**Comment by yihong0618** [comment]
will fix the change in python3.8

File: .github/workflows/ci.yml (hunk 1 above)
Comment (matin) [reply .github/workflows/ci.yml:22]: There are open issues with 3.8:
```
FAILED tests/stats/test_intensity_minutes.py::test_weekly_intensity_minutes - AttributeError: 'tuple' object has no attribute 'week'
FAILED tests/stats/test_steps.py::test_weekly_steps - TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
FAILED tests/stats/test_stress.py::test_daily_stress - TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
FAILED tests/stats/test_stress.py::test_daily_stress_pagination - TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
```

We can make the necessary changes to support 3.8, but I recommend doing so outside of this PR.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (matin) [reply .github/workflows/ci.yml:13]: Didn't know this. Thanks!

**Comment by matin** [comment]
@yihong0618 with the exception of 3.8, looks great!

Could you remove 3.8 support? It can always be resolved outside of this PR.

**Comment by yihong0618** [comment]
of course, dropping it

**Comment by matin** [comment]
@yihong0618 published as version `0.4.28`

**Pull request merged**